CatalogAPI/Repository/CategoryRepository.cs

```
@@ -0,0 +1,92 @@
+﻿using CatalogAPI.Data;
+using CatalogAPI.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace CatalogAPI.Repository
+{
+    public class CategoryRepository : ICategoryRepository
+    {
+        private readonly CatalogContext _context;
+
+        public CategoryRepository(CatalogContext context)
+        {
+            _context = context;
+        }
+        public async Task<List<CategoryModel>> GetAllCategoriesAsync()
+        {
+            var results = await _context.Category.Select(x=> new CategoryModel
+            {
+                Id = x.Id,
+                Name = x.Name                    
+            }).ToListAsync();
+            return results;
+        }
+
+        public async Task<CategoryModel> GetCategoryById(long id)
+        {
+            if(id<0)
+            {
+                throw new ArgumentNullException(nameof(id), "Id should be equal or greater than 0!");
+            }
+            var result = await _context.Category.Where(x => x.Id == id).Select(x => new CategoryModel
+            {
+                Id = x.Id,
+                Name = x.Name
+            }).FirstOrDefaultAsync();
+            if (result == null)
```

Not the right exception to use.
Also, I would let the null propagate down to the controller and return NotFound(message) there if it was null.